src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -133,6 +138,162 @@ public ObservableList<Person> getInternalList() {
         return internalList;
     }
 
+    /**
+     * Returns the next first available class.
+     *
+     * @param tr a timeRange object containing the {@code startTime}, {@code endTime} and {@code duration}.
+     * @return the next first available Class.
+     */
+    public Class getAvailableClass(TimeRange tr) {
+        LocalDate currDate = LocalDate.now();
+        List<Person> list = internalList
+                .stream()
+                .filter(person -> person.getAClass().startTime != null
+                        && person.getAClass().endTime != null
+                        && person.getAClass().date != null
```

Comment:
I realised this wouldn't work because it would neglect the initial slot.

So for example, you have 2 classes at `0000-0100` and `0900-1000`. It is currently 0142 HRS, so if you run `avail 0000-1200 60`, you will get the `0000-0100` slot which is wrong.

Reply:
Why this is wrong tho? If we claim that we only find the available timeslot on the NEXT day since today's schedule can be easily seen from right schedule panel.

Reply:
> @Echomo-Xinyu I realised this wouldn't work because it would neglect the initial slot.
> 
> So for example, you have 2 classes at `0000-0100` and `0900-1000`. It is currently 0142 HRS, so if you run `avail 0000-1200 60`, you will get the `0000-0100` slot which is wrong.

Why this is wrong tho? If we claim that we only find the available timeslot on the NEXT day since today's schedule can be easily seen from right schedule panel.

Reply:
@Echomo-Xinyu  It is wrong because it will filter out the `0000-0100` slot, and it will take the `0000-0100` slot on the same day

Reply:
But you are no longer looking at today's lesson right? So there is no need to find available timeslots from 01:42 onwards.

Reply:
After discussion, we have reached the conclusion that we are trying to find the available timeslot in today as well. So the discussion above becomes no longer valid.